### PR TITLE
account for multiple mgrs in the storage system

### PR DESCRIPTION
**Pull request by weirdwiz**
https://bugzilla.redhat.com/show_bug.cgi?id=2275965
https://bugzilla.redhat.com/show_bug.cgi?id=2272928
loop over all the results in PrometheusResult

image

### CODE REVIEW

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx [hunk 1]
```diff
@@ -51,20 +50,26 @@ const calculateDaysUp = (response: PrometheusResponse): number | null => {
...
+    }
 
-  const timestamps: number[] = values.map(([timestamp, _value]) => timestamp);
+    const timestamps: number[] = promResult.values.map(
+      ([timestamp, _value]) => timestamp
+    );
 
-  const minTimestamp: number = Math.min(...timestamps);
-  const maxTimestamp: number = Math.max(...timestamps);
+    minTimestamp = Math.min(...timestamps, minTimestamp ?? timestamps[0]);
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:66]: nits: why not just default `minTimestamp` to `0` like `maxTimestamp` ?
```suggestion
    minTimestamp = Math.min(...timestamps, minTimestamp);
```

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx [hunk 2]
```diff
@@ -51,20 +50,26 @@ const calculateDaysUp = (response: PrometheusResponse): number | null => {
     return null;
   }
 
-  const values: PrometheusValue[] = response.data?.result?.[0]?.values || [];
+  let minTimestamp: number;
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:53]: https://github.com/red-hat-storage/odf-console/pull/1312/files#r1571917458

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx (hunk 1 above)
Comment (weirdwiz) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:66]: 0 is the least value, if se set min to that. it'll always be least

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:66]: lol, my bad...

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx [hunk 3]
```diff
@@ -51,20 +50,26 @@ const calculateDaysUp = (response: PrometheusResponse): number | null => {
...
+    );
 
-  const minTimestamp: number = Math.min(...timestamps);
-  const maxTimestamp: number = Math.max(...timestamps);
+    minTimestamp = Math.min(...timestamps, minTimestamp ?? timestamps[0]);
+    maxTimestamp = Math.max(...timestamps, maxTimestamp);
+  }
 
-  const timeDifferenceMs: number = maxTimestamp - minTimestamp;
+  const timeDifferenceSecond: number = maxTimestamp - minTimestamp;
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:70]: for edge case if `promResult.values.length` is empty...
```suggestion
  const timeDifferenceSecond: number = maxTimestamp - (minTimestamp || 0);
```

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:66]: https://github.com/red-hat-storage/odf-console/pull/1312/files#r1571929991

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx (hunk 3 above)
Comment (weirdwiz) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:70]: if length is zero, the total value will be 0 (0-null) is zero from what i can see on the console

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:53]: plz ignore...

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx (hunk 3 above)
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:70]: `0-undefined` is `NaN`

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:70]: most commonly u will find `undefined` with JS...

Comment (weirdwiz) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx]: updated

**Comment by SanjalKatiyar** [comment]
/approve cancel
cc @bipuladh for approval

**Pull request merged**